mlflow/genai/scorers/deepeval/utils.py

```diff
@@ -0,0 +1,181 @@
+    extract_retrieval_context_from_trace,
+    parse_outputs_to_str,
+    resolve_expectations_from_trace,
+    resolve_inputs_from_trace,
+    resolve_outputs_from_trace,
+)
+from mlflow.protos.databricks_pb2 import BAD_REQUEST
+
+DEEPEVAL_NOT_INSTALLED_ERROR_MESSAGE = (
+    "DeepEval metrics require the 'deepeval' package. Please install it with: pip install deepeval"
+)
+
+
+def _check_deepeval_installed():
+    try:
+        import deepeval  # noqa: F401
+    except ImportError:
+        raise MlflowException(
+            DEEPEVAL_NOT_INSTALLED_ERROR_MESSAGE,
+            error_code=BAD_REQUEST,
+        )
+
+
+def create_deepeval_model(model_uri: str):
+    _check_deepeval_installed()
+    from deepeval.models import LiteLLMModel
+
+    if model_uri == "databricks":
+        from mlflow.genai.scorers.deepeval.models import DatabricksDeepEvalLLM
+
+        return DatabricksDeepEvalLLM()
+    elif model_uri.startswith("databricks:/"):
+        from mlflow.genai.scorers.deepeval.models import DatabricksServingEndpointDeepEvalLLM
+
+        endpoint_name = model_uri.split(":", 1)[1].lstrip("/")
+        return DatabricksServingEndpointDeepEvalLLM(endpoint_name)
+    elif ":" in model_uri:
+        provider, model_name = model_uri.split(":", 1)
+        model_name = model_name.lstrip("/")
+        return LiteLLMModel(model=f"{provider}/{model_name}")
+
+    raise MlflowException(
+        (
+            f"Invalid model URI: '{model_uri}'. Expected format: 'databricks', "
+            "'databricks:/<endpoint_name>', or a LiteLLM model URI."
+        ),
+        error_code=BAD_REQUEST,
+    )
+
+
+def _convert_to_deepeval_tool_calls(tool_call_dicts: list[dict[str, Any]]):
+    """
+    Convert tool call dicts to DeepEval ToolCall objects.
+
+    Args:
+        tool_call_dicts: List of dicts with tool call data
+
+    Returns:
+        List of DeepEval ToolCall objects
+    """
+    try:
+        from deepeval.test_case import ToolCall as DeepEvalToolCall
+    except ImportError as e:
+        raise MlflowException(
+            DEEPEVAL_NOT_INSTALLED_ERROR_MESSAGE,
+            error_code=BAD_REQUEST,
+        ) from e
+
+    tool_calls = []
+    for tc_dict in tool_call_dicts:
+        tool_call = DeepEvalToolCall(
+            name=tc_dict.get("name"),
+            description=tc_dict.get("description"),
+            reasoning=tc_dict.get("reasoning"),
+            output=tc_dict.get("output"),
+            input_parameters=tc_dict.get("input_parameters"),
+        )
+        tool_calls.append(tool_call)
+
+    return tool_calls
+
+
+def _extract_tool_calls_from_trace(trace: Trace):
+    """
+    Extract tool calls from trace spans with type TOOL.
+
+    Args:
+        trace: MLflow Trace object
+
+    Returns:
+        List of DeepEval ToolCall objects, or None if no tool calls found
+    """
+    if not trace:
+        return None
+
+    try:
+        from deepeval.test_case import ToolCall as DeepEvalToolCall
+    except ImportError as e:
```

Comment:
I see similar try catch in several places, can we simplify them? I think we don't need try catch everywhere, only important upstream should check, and if it fails these functions won't be invoked